test/e2e/suites/base/clustertaintpolicy_test.go

```diff
@@ -0,0 +1,267 @@
+/*
+Copyright 2025 The Karmada Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package base
+
+import (
+	"fmt"
+
+	"github.com/onsi/ginkgo/v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	clusterv1alpha1 "github.com/karmada-io/karmada/pkg/apis/cluster/v1alpha1"
+	policyv1alpha1 "github.com/karmada-io/karmada/pkg/apis/policy/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+	"github.com/karmada-io/karmada/test/e2e/framework"
+	karmadaresource "github.com/karmada-io/karmada/test/e2e/framework/resource/karmada"
+)
+
+var _ = framework.SerialDescribe("Taint cluster with ClusterTaintPolicy", func() {
+	var policyName string
+	var clusterTaintPolicy *policyv1alpha1.ClusterTaintPolicy
+	var targetClusterName string
+
+	ginkgo.JustBeforeEach(func() {
+		karmadaresource.CreateClusterTaintPolicy(karmadaClient, clusterTaintPolicy)
+		ginkgo.DeferCleanup(func() {
+			karmadaresource.RemoveClusterTaintPolicy(karmadaClient, policyName)
+		})
+	})
+
+	ginkgo.BeforeEach(func() {
+		policyName = clusterTaintPolicyNamePrefix + rand.String(RandomStrLength)
+		clusterTaintPolicy = &policyv1alpha1.ClusterTaintPolicy{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: policyv1alpha1.SchemeGroupVersion.String(),
+				Kind:       "ClusterTaintPolicy",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: policyName,
+			},
+		}
+	})
+
+	ginkgo.Context("One condition match", func() {
+		ginkgo.BeforeEach(func() {
+			targetClusterName = framework.ClusterNames()[1]
+			clusterTaintPolicy.Spec = policyv1alpha1.ClusterTaintPolicySpec{
+				TargetClusters: &policyv1alpha1.ClusterAffinity{
+					ClusterNames: []string{targetClusterName},
+				},
+				AddOnConditions: []policyv1alpha1.MatchCondition{
+					{
+						ConditionType: "NetworkReady",
+						Operator:      policyv1alpha1.MatchConditionOpIn,
+						StatusValues:  []metav1.ConditionStatus{metav1.ConditionFalse, metav1.ConditionUnknown},
+					},
+				},
+				RemoveOnConditions: []policyv1alpha1.MatchCondition{
+					{
+						ConditionType: "NetworkReady",
+						Operator:      policyv1alpha1.MatchConditionOpIn,
+						StatusValues:  []metav1.ConditionStatus{metav1.ConditionTrue},
+					},
+				},
+				Taints: []policyv1alpha1.Taint{
+					{
+						Key:    "testing/network-not-ready",
+						Effect: "NoSchedule",
```

Comment:
Is it mandatory to use a real `effect`? 
If we use a dummy effect here, this test can be run in parallel, right?

Reply:
Yes, if we use a dummy effect taint, it can be executed in parallel. However, the value of effect will be checked later, and serialization is still required.